test/issues/292.js

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2018 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it before after */
+'use strict';
+
+const assert = require('assert');
+const Client = require('../../lib/client.js');
+const Cloudant = require('../../cloudant.js');
+const nock = require('../nock.js');
+const uuidv4 = require('uuid/v4'); // random
+
+const ME = process.env.cloudant_username || 'nodejs';
+const PASSWORD = process.env.cloudant_password || 'sjedon';
+const SERVER = `https://${ME}.cloudant.com`;
+const DBNAME = `nodejs-cloudant-${uuidv4()}`;
+
+describe('Issue #292', function() {
```

Comment:
Wouldn't it be better to put these tests under a file named like `index_with_promise_tests.js`? This way we could put any future index tests in the same location.

Reply:
These tests are specific to the issue. I'd prefer index tests to live in the `cloudant.js` test file. Note there are no index tests currently.

Reply:
Ok sounds good.